climada/hazard/tc_tracks_synth.py

```diff
@@ -204,6 +196,21 @@ def calc_perturbed_trajectories(tracks,
                                  max_dspeed_rel, max_ddirection, rand)
                    for track, rand in zip(tracks.data, random_vec)]
 
+    cutoff_track_ids_cat = [x[1] for x in new_ens]
+    cutoff_track_ids_cat = sum(cutoff_track_ids_cat, [])
+    cutoff_track_ids_ok = [x[2] for x in new_ens]
+    cutoff_track_ids_ok = sum(cutoff_track_ids_ok, [])
+    if len(cutoff_track_ids_cat) > 0:
+        LOGGER.warning('The following generated synthetic tracks moved beyond '
+                        'the range of [-70, 70] degrees latitude. Cut out '
+                        'at TC category >1: %s.',
+                        ', '.join(cutoff_track_ids_cat))
+    if len(cutoff_track_ids_ok) > 0:
+        LOGGER.info('The following generated synthetic tracks moved beyond '
+                    'the range of [-70, 70] degrees latitude. Cut out '
+                    'at TC category <= 1: %s.',
+                    ', '.join(cutoff_track_ids_ok))
+    new_ens = [x[0] for x in new_ens]
```

Comment:
I am not really convinced that this is actually needed. In any case, when computing windfields, there is a hard cut at 60°. I think this is unecessary 'false' precision.

Reply:
> Thanks @chahank I'll try to implement the tests.
> In the meantime note that I now allow single point tracks to be perturbed (it's just the initial perturbation that is applied then). Do you want me to revert the changes to read_ibtracs_netcdf or should I keep that option of skipping single-point tracks?
> I've also reformatted the code to use TC category as keys to dictionary consistently with `climada.hazard.tc_tracks.CAT_NAMES`, and use `climada.hazard.tc_tracks.set_category` instead of a manual calculation in the code.

We are getting there! I think it is fine to leave the changes to `read_ibtracs_netcdf`. The reformating for the TC categories is very helpful, thanks! 

Beyond the small points noted in the last changes, the only thing missing are the unit tests.